source/linux/Foundational_Components/Graphics/Rogue/Build_Guide.rst

```diff
@@ -0,0 +1,362 @@
+###########
+Build Guide
+###########
+
+The graphics stack on rogue has fluctuated a little over the years as we've
+adjusted for binary longevity in the proprietary stack. This will attempt to go
+over some of the discrepancies between releases and how you can build the
+releases stand alone.
+
+*****************
+Proprietary Stack
+*****************
+
+The proprietary stack has been our default offering for all releases so far. It
+consists of 3 main components with varying classifications:
+
+   - The kernel module (KM) (Open Source, but out-of-tree)
+   - The GLES / Vulkan / OpenCL implementation (UM) (Closed Source, binary release)
+   - The mesa shim (Open Source, but out-of-tree)
+
+All 3 of these components are required for GLES and Vulkan support. Exceptions
+will be covered in :ref:`rogue-proprietary-mesa-components`.
+
+Proprietary Kernel Module
+=========================
+
+The kernel module is required for:
+
+   - Memory and power management
+   - Loading firmware
+   - Monitoring device state
+   - Orchestrating hardware recoveries
+   - Creating an interface for the userspace server
+
+The source for the component is available at:
+
+   https://git.ti.com/cgit/graphics/ti-img-rogue-driver/
+
+.. _rogue-building-the-proprietary-kernel-module:
+
+Building the Proprietary Kernel Module
+--------------------------------------
+
+Pick a branch that corresponds to your particular kernel. Keep note of the name.
+The UM component must match this release.
+
+The :file:`INSTALL` file gives a brief overview of how to build and install the
+module. This gives you the common variables, but it does leave out some of the
+more important ones for the build. The following variables are key and *must*
+match with the values used in the UM release:
+
+   - ``BUILD`` -- Options: ``release`` or ``debug``
+   - ``PVR_BUILD_DIR`` -- Options: ``<platform>_linux``, for example ``am62_linux``
+   - ``WINDOW_SYSTEM`` -- Options: ``lws-generic``, ``xorg``, ``wayland``, ...
+
+These variables must be set for the build. To check for updated build variables
+please see the ``EXTRA_OEMAKE`` variable in the `ti-img-rogue-driver recipe in
+meta-ti`_. This will always contain the latest required variables to compile the
+driver.
+
+.. _ti-img-rogue-driver recipe in meta-ti: https://git.ti.com/cgit/arago-project/meta-ti/tree/meta-ti-bsp/recipes-bsp/powervr-drivers/ti-img-rogue-driver_24.1.6554834.bb
```

Comment:
Same comment as https://github.com/TexasInstruments/processor-sdk-doc/pull/57#discussion_r1867152237

Reply:
This one will change eventually, but I intent to point this to the latest version on each release.

Reply:
The idea being, this section should be checked for updates upon each release anyway.

Reply:
Great, that you are planning to keep it updated in each release.

My concern was on SDK docs of older version (they are older to us but still relevant to users) which would eventually point to broken link once the recipe name has an update.

Better to use tag as well in such links?

Probably - https://git.ti.com/cgit/arago-project/meta-ti/tree/meta-ti-bsp/recipes-bsp/powervr-drivers/ti-img-rogue-driver_24.1.6554834.bb?h=10.01.09

Reply:
Ah, yeah. Fair point.